src/main/java/com/google/chcapi/perfdiag/profiler/HttpRequestProfilerFactory.java

```diff
@@ -201,23 +202,39 @@ public static HttpRequestProfiler createRetrieveDicomInstanceFrameRequest(
    * @param download {@code true} if it is download request.
    * @return A new prepared HTTP GET request instance.
    */
-  private static HttpUriRequest createHttpGetRequest(String uri, boolean download) {
+  private static HttpUriRequest createHttpGetRequest(String uri, boolean withDicomMultipartHeader) {
     final HttpGet request = new HttpGet(uri);
-    if (download) {
+    if (withDicomMultipartHeader) {
       request.setHeader("Accept", "multipart/related; type=application/dicom; transfer-syntax=*");
     }
     request.setHeader("Authorization", "Bearer " + CREDENTIAL.getAccessToken());
     return request;
   }
 
+  /**
+   * Constructs a new HTTP GET request for the specified URI.
+   *
+   * @param uri The HTTP request URI.
+   * @param download {@code true} if it is download request.
+   * @return A new prepared HTTP GET request instance.
+   */
+  public static HttpRequestProfiler createQidoRequest(DicomWebRequestConfig config) {
+    return new HttpRequestProfiler(
+            createHttpGetRequest(
+                    buildDicomWebURI(config, false)
+                            .append(config.getRequestPath())
+                            .toString(),
+                    false));
+  }
+
   /**
    * Constructs DICOM Web request URI using parameters from the specified DICOM store configuration.
    *
    * @param config DICOM store configuration.
    * @return DICOM Web request URI as {@code StringBuilder} instance for further URI construction.
    */
-  private static StringBuilder buildDicomWebURI(DicomStoreConfig config) {
-    return new StringBuilder(endpoint)
+  private static StringBuilder buildDicomWebURI(DicomStoreConfig config, boolean withTrailingStudies) {
```

Comment:
Instead of a bool here that isn't obvious when you're reading the code, can you just fix all previous usages of this method to append "/studies" themselves?